bindings/python/convert.py

```diff
@@ -14,59 +31,95 @@ def rename(pt_filename) -> str:
-            os.remove(local)
+def convert_multi(model_id, folder):
+    filename = hf_hub_download(repo_id=model_id, filename="pytorch_model.bin.index.json")
+    with open(filename, "r") as f:
+        data = json.load(f)
 
+    filenames = set(data["weight_map"].values())
+    local_filenames = []
+    for filename in filenames:
+        cached_filename = hf_hub_download(repo_id=model_id, filename=filename)
+        loaded = torch.load(cached_filename)
+        sf_filename = rename(filename)
 
-def convert_single(model_id):
-    local = "model.safetensors"
-    try:
-        filename = hf_hub_download(repo_id=model_id, filename="pytorch_model.bin")
-        loaded = torch.load(filename)
+        local = os.path.join(folder, sf_filename)
         save_file(loaded, local, metadata={"format": "pt"})
+        check_file_size(local, cached_filename)
+        local_filenames.append(local)
 
-        api = HfApi()
+    index = os.path.join(folder, "model.safetensors.index.json")
+    with open(index, "w") as f:
+        newdata = {k: v for k, v in data.items()}
+        newmap = {k: rename(v) for k, v in data["weight_map"].items()}
+        newdata["weight_map"] = newmap
+        json.dump(newdata, f)
+    local_filenames.append(index)
 
-        api.upload_file(
-            path_or_fileobj=local,
-            create_pr=True,
-            path_in_repo=local,
-            repo_id=model_id,
-        )
+    operations = [
+        CommitOperationAdd(path_in_repo=local.split("/")[-1], path_or_fileobj=local) for local in local_filenames
+    ]
+
+    return operations
+
+
+def convert_single(model_id, folder):
+    sf_filename = "model.safetensors"
+    filename = hf_hub_download(repo_id=model_id, filename="pytorch_model.bin")
+    loaded = torch.load(filename)
+
+    local = os.path.join(folder, sf_filename)
+    save_file(loaded, local, metadata={"format": "pt"})
+
+    check_file_size(local, filename)
+
+    operations = [CommitOperationAdd(path_in_repo=sf_filename, path_or_fileobj=local)]
+    return operations
+
+
+def check_final_model(model_id, folder):
+    config = hf_hub_download(repo_id=model_id, filename="config.json")
+    shutil.copy(config, os.path.join(folder, "config.json"))
+    config = AutoConfig.from_pretrained(folder)
+    _, sf_model = infer_framework_load_model(folder, config)
+    _, pt_model = infer_framework_load_model(model_id, config)
+
+    input_ids = torch.arange(10).long().unsqueeze(0)
+    sf_logits = sf_model(input_ids)
+    pt_logits = pt_model(input_ids)
+    torch.testing.assert_close(sf_logits, pt_logits)
+    print(f"Model {model_id} is ok !")
+
+
+def convert(api, model_id):
+    info = api.model_info(model_id)
+    filenames = set(s.rfilename for s in info.siblings)
+
+    folder = repo_folder_name(repo_id=model_id, repo_type="models")
+    os.makedirs(folder)
+    new_pr = None
+    try:
+        operations = None
+        if "model.safetensors" in filenames or "model_index.safetensors.index.json" in filenames:
```

Comment:
maybe at some point we'll want a `--force` flag (to correct already-merged incorrect safetensors weights for instance)

Reply:
Good suggestion !